### PR TITLE
Itext selection on canvas

### PR DESCRIPTION
**Pull request by asturur**
This PR allows the user to select text moving mouse outside the iText object bounding box

@kangax i don't get if i have to de-register the event on mouseup or on exit editing or if is not necessary.
Is not clear to me.

### CODE REVIEW

File: src/mixins/itext_behavior.mixin.js [hunk 1]
```diff
@@ -341,10 +341,28 @@
       this._setEditingProps();
 
       this._tick();
-      this.canvas && this.canvas.renderAll();
-
       this.fire('editing:entered');
-      this.canvas && this.canvas.fire('text:editing:entered', { target: this });
+      if (this.canvas) {
```
Comment (kangax) [inline src/mixins/itext_behavior.mixin.js]: Hm, why not have it in a separate method?

Comment (asturur) [inline src/mixins/itext_behavior.mixin.js]: you mean the "function(options) {}" ?

i just moved it from other section. i was enough happy that it was already working.

anyway an early return is something to do if !this.canvas

Comment (kangax) [inline src/mixins/itext_behavior.mixin.js]: Yes, I thought about early return at first, coz it's simple and makes it easier to read afterwards. But then I saw you removed `initMousemoveHandler` and I realized that having all that code in a separate function is even better.

**Comment by asturur** [comment]
@kangax i restored the method. i didn't the meaning of your comment at first.

**Comment by kangax** [comment]
Great, thanks

**Comment by asturur** [comment]
But should we not deregister the event? on exitEdit ? Does not the mouse:move stack one over the other? First the event was registered at element creation, now is on element enterEdit, so multiple times.

**Comment by kangax** [comment]
Updated fabricjs.com with latest changes but it doesn't seem to work for me — 
http://fabricjs.com/test/misc/itext.html

**Comment by kangax** [comment]
You're right we should. Or it'll be adding up infinitely.

**Comment by asturur** [comment]
on kitchensing is fine
https://www.youtube.com/watch?v=vL8IbuWyBqo
For derigisted i can use this.canvas.off, but how do i get a pointer to that event?

**Comment by kangax** [comment]
It works now! Maybe needed time to update on a server.

**Pull request merged**